### PR TITLE
Gnosis Support

### PR DESCRIPTION
**Pull request by manuelsc**
Implements feature request of issue #92 

- Support Gnosis Network (Widgets, abstract support for different EL and CL currencies, free Gnosis theme)
- Validator status now uses beaconcha.in api instead of calculating it based on recent epoch data
- Refactored api service to always use native requests
- Refactored unit conversion service, fixed finney convertion bug
- Preload blocks for faster results when viewing something in the blocks tab
- Fixed client update re-appearing multiple times without change
- Fixed wrong RPL collateral percentage
- Loosen restriction regarding non play services androids so they can configure notifications with a note that they won't work (but can be set up via webhooks) - #231 
- Speed and memory usage optimizations

### CODE REVIEW

**Review by D13ce — CHANGES_REQUESTED** [review]

File: src/app/components/help/help.component.ts [hunk 1]
```diff
@@ -35,12 +42,33 @@ export class HelpComponent implements OnInit {
 	@Input() onlyGuides: boolean
 	isAlreadyLoggedIn = false
 
-	constructor(private oauthUtils: OAuthUtils, private validator: ValidatorUtils, private storage: StorageService, private router: Router) {}
+	isGnosis: boolean
+
+	private ethereumNetworkKey: string
```
[the remark this answers is not in the document]
Comment (manuelsc) [reply src/app/components/help/help.component.ts:47]: Imagine a scenario where the user is currently on an Ethereum testnet, fe goerli. isGnosis() is false and we switch to gnosis. Now when the user presses the button, we want to switch back to goerli not main. Hence the introduction of ethereumNetworkKey

File: src/app/components/dashboard/dashboard.component.ts
[the remark this answers is not in the document]
Comment (manuelsc) [reply src/app/components/dashboard/dashboard.component.ts:1]: I'm not sure if this is a rounding error. I think the data via the API is just newer, probably best to ask someone from the explorer team how does values appear on the frontend.

File: src/app/services/api.service.ts [hunk 2]
```diff
@@ -183,28 +170,29 @@ export class ApiService extends CacheModule {
...
-
+	private unlock(resource) {
 		this.awaitingResponses[resource].release()
 	}
 
-	async isNotMainnet(): Promise<boolean> {
-		const test = (await this.networkConfig).net != ''
+	isNotEthereumMainnet(): boolean {
+		const test = this.networkConfig.net != ''
 		return test
```
Comment (D13ce) [inline src/app/services/api.service.ts:184]: **Bug:** This check returns true for Gnosis as well.

**Review by D13ce — CHANGES_REQUESTED** [review]

File: src/app/services/api.service.ts (hunk 2 above)
Comment (manuelsc) [reply src/app/services/api.service.ts:184]: This is okay, I renamed it to isNotEthereumMainnet

Comment (D13ce) [reply src/app/services/api.service.ts:184]: Sorry, I meant this returns false for Gnosis as well i.e. Gnosis is handled as Ethereum's mainnet by this function.

Comment (D13ce) [reply src/app/services/api.service.ts:184]: Has been solved here: https://github.com/gobitfly/eth2-beaconchain-explorer-app/pull/255/files#diff-654ea4aabcdd84b489045ca408d33a859f0f18c051ec3cb15801baca9cf3919aR183

File: src/app/components/dashboard/dashboard.component.ts
[the remark this answers is not in the document]
Comment (D13ce) [reply src/app/components/dashboard/dashboard.component.ts:1]: Seems to be fixed via https://github.com/gobitfly/eth2-beaconchain-explorer-app/pull/255 .

[the remark this answers is not in the document]
Comment (D13ce) [reply src/app/components/dashboard/dashboard.component.ts:1]: Seems to be fixed in https://github.com/gobitfly/eth2-beaconchain-explorer-app/pull/255 .

[the remark this answers is not in the document]
Comment (D13ce) [reply src/app/components/dashboard/dashboard.component.ts:1]: Seems to be fixed in https://github.com/gobitfly/eth2-beaconchain-explorer-app/pull/255 .

**Review by D13ce — DISMISSED** [review]
All remaining issues seem to be fixed with https://github.com/gobitfly/eth2-beaconchain-explorer-app/pull/255 .

lgtm

**Review by D13ce — APPROVED** [review]
lgtm :tada:

**Pull request merged**